server/src/test/java/com/objectcomputing/checkins/services/template_question/TemplateQuestionControllerTest.java

```diff
@@ -0,0 +1,396 @@
+                .basicAuth(memberOne.getWorkEmail(), RoleType.Constants.MEMBER_ROLE);
+        final HttpResponse<TemplateQuestionResponseDTO> response2 = client.toBlocking().exchange(request2, TemplateQuestionResponseDTO.class);
+
+        assertEquals(HttpStatus.OK, response1.getStatus());
+        assertTrue(response1.getBody().isPresent());
+        assertContentEqualsResponse(question1, response1.getBody().get());
+
+        assertEquals(HttpStatus.OK, response2.getStatus());
+        assertTrue(response2.getBody().isPresent());
+        assertContentEqualsResponse(question2, response2.getBody().get());
+    }
+
+    @Test
+    void testGetByIdUnauthorized() {
+        final MemberProfile memberOne = createADefaultMemberProfile();
+
+        final FeedbackTemplate template = saveDefaultFeedbackTemplate(memberOne.getId());
+        final TemplateQuestion question1 = saveQuestion("How are you today?", template.getId(), 1);
+
+        final HttpRequest<?> request = HttpRequest.GET(String.format("%s", question1.getId()));
+        final HttpClientResponseException exception = assertThrows(HttpClientResponseException.class,
+                () -> client.toBlocking().exchange(request, Map.class));
+
+        final JsonNode body = exception.getResponse().getBody(JsonNode.class).orElse(null);
+
+        assertNull(body);
+        assertEquals(HttpStatus.UNAUTHORIZED, exception.getStatus());
+        assertEquals("Unauthorized", exception.getMessage());
+    }
+
+    @Test
+    void testGetByIdNotFound() {
+        final MemberProfile memberOne = createADefaultMemberProfile();
+
+        final FeedbackTemplate template = saveDefaultFeedbackTemplate(memberOne.getId());
+        final TemplateQuestion question1 = saveQuestion("How are you today?", template.getId(), 1);
+        final UUID nonExistentId = UUID.randomUUID();
+
+        final HttpRequest<?> request = HttpRequest.GET(String.format("%s", nonExistentId))
+                .basicAuth(memberOne.getWorkEmail(), RoleType.Constants.MEMBER_ROLE);
+        final HttpClientResponseException exception = assertThrows(HttpClientResponseException.class,
+                () -> client.toBlocking().exchange(request, Map.class));
+
+        assertEquals("No feedback question with ID " + nonExistentId, exception.getMessage());
+        assertEquals(HttpStatus.NOT_FOUND, exception.getStatus());
+    }
+
+    @Test
+    void testUpdateQuestionAndOrderAuthorized() {
+        final MemberProfile memberOne = createADefaultMemberProfile();
+        FeedbackTemplate template = saveDefaultFeedbackTemplate(memberOne.getId());
+        final TemplateQuestion question1 = saveQuestion("How are you today?", template.getId(), 1);
+        final TemplateQuestionUpdateDTO updateDTO = new TemplateQuestionUpdateDTO();
+
+        // Update by the creator
+        updateDTO.setId(question1.getId());
+        updateDTO.setQuestion("Do you think opossums are misunderstood creatures?");
+        updateDTO.setOrderNum(2);
```

Comment:
We'll probably want to give this `orderNum` a uniqueness constraint in both our validations and in the database. Thoughts?

Reply:
Can be circumvented by assigning order numbers on the backend